compiler/src/main/java/org/qbicc/graph/atomic/AccessModes.java

```diff
@@ -0,0 +1,190 @@
+ * <tr><td align=center>{@link #SinglePlain}</td><td align=center>●</td><td align=center>●</td><td align=center></td><td align=center></td><td align=center></td><td align=center></td><td align=center></td>
+ * <tr><td align=center>{@link #SingleOpaque}</td><td align=center>●</td><td align=center>●</td><td align=center>●</td><td align=center></td><td align=center></td><td align=center></td><td align=center></td>
+ * <tr><td align=center>{@link #SingleAcquire}</td><td align=center>●</td><td align=center>●</td><td align=center>●</td><td align=center>●</td><td align=center></td><td align=center></td><td align=center></td>
+ * <tr><td align=center>{@link #SingleRelease}</td><td align=center>●</td><td align=center>●</td><td align=center>●</td><td align=center></td><td align=center>●</td><td align=center></td><td align=center></td>
+ * <tr><td align=center>{@link #SingleAcqRel}</td><td align=center>●</td><td align=center>●</td><td align=center>●</td><td align=center>●</td><td align=center>●</td><td align=center>●</td><td align=center></td>
+ * <tr><td align=center>{@link #SingleSeqCst}</td><td align=center>●</td><td align=center>●</td><td align=center>●</td><td align=center>●</td><td align=center>●</td><td align=center>●</td><td align=center>●</td>
+ * </table>
+ * <p>
+ * Apart from {@link #SingleAcquire} and {@link #SingleRelease} — which do not include one another — every mode includes
+ * all of the preceding modes.
+ * <p>
+ * Additionally, every single-access mode can be strengthened into a corresponding global access mode which includes at
+ * least the behavior of the single-access mode when applied to the same memory operation.
+ * <p>
+ * <table class="striped">
+ * <tr><th align=center>Single Access Mode</th><th align=center>Global Access mode</th></tr>
+ * <tr><td align=center>{@link #SingleUnshared}</td><td align=center>{@link #GlobalSeqCst}</td>
+ * <tr><td align=center>{@link #SinglePlain}</td><td align=center>{@link #GlobalSeqCst}</td>
+ * <tr><td align=center>{@link #SingleOpaque}</td><td align=center>{@link #GlobalSeqCst}</td>
+ * <tr><td align=center>{@link #SingleAcquire}</td><td align=center>{@link #GlobalAcquire}</td>
+ * <tr><td align=center>{@link #SingleRelease}</td><td align=center>{@link #GlobalRelease}</td>
+ * <tr><td align=center>{@link #SingleAcqRel}</td><td align=center>{@link #GlobalAcqRel}</td>
+ * <tr><td align=center>{@link #SingleSeqCst}</td><td align=center>{@link #GlobalSeqCst}</td>
+ * </table>
+ * <p>
+ */
+public final class AccessModes {
+    private AccessModes() {}
+
+    /**
+     * The <em>load-load</em> global access mode.  When used as a fence, this mode prevents any
+     * loads which succeed the fence from being moved before any global loads which precede the fence in the original program order:
+     * {@code AnyLoad ≺ GlobalLoadLoad < AnyGlobalLoad}.
+     */
```

Comment:
I think I may have these definitions inverted or am missing the nuance of preventing preceding operations from moving past the fence to a succeeding position.

Reply:
Fixed... I think!